Source/Evolve/WarpXComputeDt.cpp

```diff
@@ -27,29 +28,30 @@
 #include <algorithm>
 #include <memory>
 
+/**
+ * Compute the minimum of array x, where x has dimension AMREX_SPACEDIM
+ */
+AMREX_FORCE_INLINE amrex::Real
+minDim (const amrex::Real* x)
+{
+    return std::min({AMREX_D_DECL(x[0], x[1], x[2])});
+}
+
 /**
  * Determine the timestep of the simulation. */
 void
 WarpX::ComputeDt ()
 {
     // Handle cases where the timestep is not limited by the speed of light
-    if (electromagnetic_solver_id == ElectromagneticSolverAlgo::None ||
-        electromagnetic_solver_id == ElectromagneticSolverAlgo::HybridPIC) {
-
+    // and no constant timestep is provided
+    if (electromagnetic_solver_id == ElectromagneticSolverAlgo::HybridPIC) {
         std::stringstream errorMsg;
-        if (electrostatic_solver_id != ElectrostaticSolverAlgo::None) {
-            errorMsg << "warpx.const_dt must be specified with the electrostatic solver.";
-        } else if (electromagnetic_solver_id == ElectromagneticSolverAlgo::HybridPIC) {
-            errorMsg << "warpx.const_dt must be specified with the hybrid-PIC solver.";
-        } else {
-            errorMsg << "warpx.const_dt must be specified when not using a field solver.";
-        }
+        errorMsg << "warpx.const_dt must be specified with the hybrid-PIC solver.";
         WARPX_ALWAYS_ASSERT_WITH_MESSAGE(m_const_dt.has_value(), errorMsg.str());
-
-        for (int lev=0; lev<=max_level; lev++) {
-            dt[lev] = m_const_dt.value();
-        }
-        return;
+    } else if (electromagnetic_solver_id == ElectromagneticSolverAlgo::None) {
+        std::stringstream errorMsg;
+        errorMsg << "warpx.const_dt must be specified with the electrostatic solver, or the warpx.dt_update_interval must be > 0.";
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(m_const_dt.has_value() || dt_update_interval.isActivated(), errorMsg.str());
```

Comment:
Usually we write this as:
```suggestion
        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
            m_const_dt.has_value() || dt_update_interval.isActivated(), 
            "warpx.const_dt must be specified with the electrostatic solver, or the warpx.dt_update_interval must be > 0."
        );
```
In this case the errorMsg was changed depending on the solver while the assert itself was the same, which is why the `std::stringstream` was used. I don't mind keeping this as is, just wanted to point it out since it currently follows a different practice from other places in WarpX.

Reply:
Thanks, good to know!